### PR TITLE
Fix installer when run with sudo

### PR DESCRIPTION
**Pull request by jfhovinne**
### What does this PR do?
- `sudo ./install.sh` may generate the following error: `sudo: PAM account management error: Permission denied` and then stop execution
- This PR fixes the issue by checking user UID

### What ticket does this PR close?
- N/A

#### Change log
- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [x] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [x] This PR does not require updating any documentation

### CODE REVIEW

**Comment by rpothier** [comment]
Hi @jfhovinne Thanks for submitting this request! Can you please amend your commit with [the DCO signoff](https://github.com/cyberark/community/blob/master/CONTRIBUTING.md#signing-off-a-commit)?

**Comment by jfhovinne** [comment]
Hi @rpothier Commit amended. Anything else blocking this PR ? Thanks.

**Comment by BradleyBoutcher** [comment]
Hi Jean,

Thanks for this PR, and thanks for signing off on the commit! Before we move forward, I wanted to clarify a few things. 
- Does running this script work for you without using `sudo`? I.E. just running `./install.sh`. It looks like the check you modified should default to using non-sudo commands if `sudo` fails. Is the error coming from our script, or from using `sudo install.sh`?
- What operating system are you running? We want to be able to reproduce this error in case we have to look into it further.

Thank you!

**Comment by jfhovinne** [comment]
Hi @BradleyBoutcher

Installing as root works. Installing as normal user with `sudo ./install.sh` fails.

This happens on a freshly provisioned Amazon Linux WorkSpaces (executing as workspace user); here is the output:

```
wget https://raw.githubusercontent.com/cyberark/summon/master/install.sh
chmod +x install.sh
sudo ./install.sh
Using version number: v0.8.3
Downloading https://github.com/cyberark/summon/releases/download/v0.8.3/summon-linux-amd64.tar.gz
Installing summon v0.8.3 into /usr/local/bin
sudo: PAM account management error: Permission denied
```

From /var/log/secure:

```
sudo: <user> : TTY=pts/0 ; PWD=/home/<user> ; USER=root ; COMMAND=./install.sh
sudo: pam_unix(sudo:session): session opened for user root by <user>(uid=0)
sudo: pam_access(sudo:account): access denied for user `root' from `pts/0'
sudo: root : PAM account management error: Permission denied ; TTY=pts/0 ; PWD=/home/<user> ; USER=root ; COMMAND=/bin/tar -C /usr/local/bin -o -zxvf /tmp/install.sh.5705/summon.tar.gz
```

**Comment by BradleyBoutcher** [comment]
@jfhovinne What happens when you run the script without sudo as a non-root user? Does it fail when it attempts to run `tar`?

**Comment by jfhovinne** [comment]
@BradleyBoutcher
```
./install.sh
Using version number: v0.8.3
Downloading https://github.com/cyberark/summon/releases/download/v0.8.3/summon-linux-amd64.tar.gz
Installing summon v0.8.3 into /usr/local/bin
[sudo] password for <user>:
Downloading https://raw.githubusercontent.com/cyberark/summon/master/script/complete_summon
/etc/bash_completion.d/complete_summon: Permission denied
```

FYI my use case is non-interactive installation.

**Comment by BradleyBoutcher** [comment]
Hey @jfhovinne,

Sorry for the delay on this. I'm going to go ahead and approve it, and I appreciate your working with us to run through some scenarios. It helps provide a clearer picture of the issue, in case someone else comes across this.

All good things,

Bradley

**Review by BradleyBoutcher — APPROVED** [review]
LGTM!

**Comment by izgeri** [comment]
@BradleyBoutcher let's run this in the pipeline to make sure the tests are passing before we merge

/approve-build 6611af1441ad07b6eb06f63cb015d274b492ed58

**Comment by izgeri** [comment]
[Pipeline](https://jenkins.conjur.net/blue/organizations/jenkins/cyberark--summon/detail/208-pr-pipeline/1/pipeline) is green!

**Pull request merged**
